PhysicsTools/PatAlgos/src/MuonMvaIDEstimator.cc

```diff
@@ -0,0 +1,108 @@
+#include "DataFormats/JetReco/interface/PFJet.h"
+#include "DataFormats/JetReco/interface/PFJetCollection.h"
+#include "DataFormats/PatCandidates/interface/Muon.h"
+#include "JetMETCorrections/JetCorrector/interface/JetCorrector.h"
+
+using namespace pat;
+using namespace cms::Ort;
+
+MuonMvaIDEstimator::MuonMvaIDEstimator(const edm::FileInPath &weightsfile) {
+  randomForest_ = std::make_unique<ONNXRuntime>(weightsfile.fullPath());
+  std::cout << randomForest_.get() << std::endl;
+}
+
+MuonMvaIDEstimator::~MuonMvaIDEstimator() {}
+
+void MuonMvaIDEstimator::fillDescriptions(edm::ConfigurationDescriptions &descriptions) {
+  // pfDeepBoostedJetTags
+  edm::ParameterSetDescription desc;
+  desc.add<edm::FileInPath>("mvaIDTrainingFile", edm::FileInPath("RecoMuon/MuonIdentification/data/mvaID.onnx"));
+  desc.add<std::vector<std::string>>("flav_names",
+                                     std::vector<std::string>{
+                                         "probBAD",
+                                         "probGOOD",
+                                     });
+
+  descriptions.addWithDefaultLabel(desc);
+}
+
+std::unique_ptr<cms::Ort::ONNXRuntime> MuonMvaIDEstimator::initializeGlobalCache(const edm::ParameterSet &iConfig) {
+  return std::make_unique<cms::Ort::ONNXRuntime>(iConfig.getParameter<edm::FileInPath>("mvaIDTrainingFile").fullPath());
+}
+void MuonMvaIDEstimator::globalEndJob(const cms::Ort::ONNXRuntime *cache) {}
+const reco::Muon::ArbitrationType arbitrationType = reco::Muon::SegmentAndTrackArbitration;
+std::vector<float> MuonMvaIDEstimator::computeMVAID(const pat::Muon &muon) const {
+  float local_chi2 = muon.combinedQuality().chi2LocalPosition;
+  float kink = muon.combinedQuality().trkKink;
+  float segment_comp = muon.segmentCompatibility(arbitrationType);
+  float n_MatchedStations = muon.numberOfMatchedStations();
+  float pt = muon.pt();
+  float eta = muon.eta();
+  float global_muon = muon.isGlobalMuon();
+  float Valid_pixel;
+  float tracker_layers;
+  float validFraction;
+  if (muon.innerTrack().isNonnull()) {
+    Valid_pixel = muon.innerTrack()->hitPattern().numberOfValidPixelHits();
+    tracker_layers = muon.innerTrack()->hitPattern().trackerLayersWithMeasurement();
+    validFraction = muon.innerTrack()->validFraction();
+  } else {
+    Valid_pixel = -99.;
+    tracker_layers = -99.0;
+    validFraction = -99.0;
+  }
+  float norm_chi2;
+  float n_Valid_hits;
+  if (muon.globalTrack().isNonnull()) {
+    norm_chi2 = muon.globalTrack()->normalizedChi2();
+    n_Valid_hits = muon.globalTrack()->hitPattern().numberOfValidMuonHits();
+  } else if (muon.innerTrack().isNonnull()) {
+    norm_chi2 = muon.innerTrack()->normalizedChi2();
+    n_Valid_hits = muon.innerTrack()->hitPattern().numberOfValidMuonHits();
+  } else {
+    norm_chi2 = -99;
+    n_Valid_hits = -99;
+  }
+  std::vector<std::string> input_names_{"float_input"};
+  std::vector<float> vars = {global_muon,
```

Comment:
```suggestion
  std::array<float, N> vars = {global_muon,
```
would be preferred

Reply:
The input variables to the ONNXRuntime::run are riquered to be allocated in a `cms::Ort::FloatArrays`, a `std::array<float, N>` does not work.